### PR TITLE
Update .rst doc files, solve some issues regarding document generation

### LINKED ISSUE #1287: Migrate all rST code blocks from :: to code-block: python
Our documentation has a mix of `::` and ` .. code-block: python`, we should migrate to all `code-block: python` so it's less confusing and more apparent when a code block is defined.

### PR DESCRIPTION
**Pull request by mesejo**
- Update documentation code blocks
- Remove duplicated reference in changelog
- Remove unused class DeprecationClient from docs
- Remove static file configuration in conf.py

Closes #1287

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -30,7 +30,8 @@ elasticsearch returns a 2XX response. Otherwise an instance of
 raised. You can see other exception and error states in :ref:`exceptions`. If
 you do not wish an exception to be raised you can always pass in an ``ignore``
 parameter with either a single status code that should be ignored or a list of
-them::
+them
+.. code-block:: python
```
Comment (sethmlarson) [inline docs/api.rst:34]: Typically there's a blank line between text and a directive.

File: docs/xpack.rst [hunk 2]
```diff
@@ -202,15 +202,3 @@ saves it to a destination index.
 .. autoclass:: TransformClient
    :members:
 
-
```
Comment (sethmlarson) [inline docs/xpack.rst:205]: Let's handle this outside this PR so I can more easily backport to 7.x

File: .gitignore [hunk 3]
```diff
@@ -15,3 +15,5 @@ coverage.xml
 nosetests.xml
 junit-*.xml
 build
+.idea/
```
Comment (sethmlarson) [inline .gitignore:18]: Can we actually change this to GitHubs Python gitignore template? Covers a lot more bases! Then add `test_elasticsearch/local.py` and `.idea` to it.

File: docs/api.rst (hunk 1 above)
Comment (mesejo) [reply docs/api.rst:35]: Done

File: docs/xpack.rst (hunk 2 above)
Comment (mesejo) [reply docs/xpack.rst]: Done

**Pull request merged**